### PR TITLE
Remove viewport_set_disable_environment in favor of viewport_set_environment_mode

### LINKED ISSUE #69122: WorldEnvironment makes adjustments to editor dropdowns and dialogs if background mode is set to Canvas
___

***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___

### Godot version

4.0-beta6

### System information

Windows 10, mobile renderer

### Issue description

When messing with the world environment node in 2D, I was adding some bloom effect to my game, and the editor itself now has bloom
image
image
image

### Steps to reproduce

Create a 2D scene
Add a world environment node
Set the Background mode to "Canvas"
Enable "Glow"
Increase both the "Strength" and "Blend" values
For best results, set "Blend mode" to "Replace"
Open up any sub menu (I discovered this by clicking to change the Blend Mode again, but the project settings is also effected by the post processing)

### Minimal reproduction project

[2DPostProcessingErrorProject.zip](https://github.com/godotengine/godot/files/10086139/2DPostProcessingErrorProject.zip)

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/69122

In 3.x this worked because every popup menu in the editor rendered to the editor viewport. So we disabled environment for that one viewport and everything worked fine. The only problem was that it remained disabled, even in the 2D editor, so the impact of the WorldEnvironment wasn't seen until running the project.

However, in 4.0 each popupmenu has its own viewport, so we would have to disable the environment for every popup in the editor. That would be much too cumbersome. 

This PR changes the API from setting a bool to setting an enum. By default Viewports inherit their value from the topmost parent. If that parent is set to default, then it is treated as having the environment not disabled. 

I will add docs if others agree that this is the proper solution

### CODE REVIEW

File: servers/rendering_server.cpp [hunk 1]
```diff
@@ -2255,6 +2255,11 @@ void RenderingServer::_bind_methods() {
 	BIND_ENUM_CONSTANT(VIEWPORT_CLEAR_NEVER);
 	BIND_ENUM_CONSTANT(VIEWPORT_CLEAR_ONLY_NEXT_FRAME);
 
+	BIND_ENUM_CONSTANT(VIEWPORT_ENVIRONMENT_ENABLED);
```
Comment (akien-mga) [inline servers/rendering_server.cpp:2258]: Make sure to bind them in the same order as the enum, so the numbers stay ordered in the docs.

**Review by akien-mga — APPROVED** [review]
Makes sense to me!

I wonder if a similar change would be needed for the disable 2D/3D methods.
BTW I see that on Viewport itself, only `disable_3d` is exposed as a property (with confusing docs that say to use `usage` instead?).

**Comment by clayjohn** [comment]
Updated and docs written!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by h0lley** [comment]
So we'd do something like this for every `MenuButton` that shouldn't have its popup menu affected by WorldEnviroment (which is probably generally the case)?

```swift
func _ready():
	RenderingServer.viewport_set_environment_mode(
		get_popup().get_viewport_rid(),
		RenderingServer.VIEWPORT_ENVIRONMENT_DISABLED)
```
Did I get that right or is there something more elegant?

**Pull request merged**